lib/common/models/collection.rb

```diff
@@ -83,7 +84,13 @@ def finder(object, search)
         predicates.all? do |operator, expected_value|
           op = OPERATIONS_MAP.fetch(operator)
           mock_comparator_object.send("#{attribute}=", expected_value)
-          actual_value.send(op, mock_comparator_object.send(attribute))
+
+          if op == :match
+            rexp = Regexp.escape(Regexp.escape(mock_comparator_object.send(attribute)))
```

Comment:
It's probably not a big deal either way. If you do it on the client side, then there might be more data transmission. If you do it on the server side, the server has to work harder to do the filtering. If it were a complicated regex, I would be more worried, but since we're double escaping it, it's basically just a simple string search, so it's not overwhelming.